### PR TITLE
[ML] fixing and unmuting testHRDSplit test

### LINKED ISSUE #32966: [CI] PainlessDomainSplitIT fails after multibucket
To recreate

```
./gradlew :x-pack:qa:ml-single-node-tests:integTestRunner -Dtests.class=org.elasticsearch.xpack.ml.transforms.PainlessDomainSplitIT -Dtests.method="testHRDSplit"
```

**Update** On master after the restructure the reproduce command is now 
```
./gradlew :x-pack:plugin:ml:qa:single-node-tests:integTestRunner -Dtests.class=org.elasticsearch.xpack.ml.transforms.PainlessDomainSplitIT -Dtests.method="testHRDSplit"
```

See failing build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request/15432/consoleFull

### PR DESCRIPTION
**Pull request by benwtrent**
This fixes the long muted testHRDSplit. Some minor adjustments for modern day elasticsearch changes :). 

The cause of the failure is that a new `by` field entering the model with an exceptionally high count does not cause an anomaly. We have since stopped combining the `rare` and `by` in this manner. New entries in a `by` field are not anomalous because we have no history on them yet. 

closes https://github.com/elastic/elasticsearch/issues/32966

### CODE REVIEW

File: x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java [hunk 1]
```diff
@@ -355,4 +366,16 @@ public void testHRDSplit() throws Exception {
             fail("Anomaly records were not found within 5 seconds");
```
Comment (droberts195) [inline x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java:366]: This `fail` dates back to when the `assertBusy` on line 330 was an `awaitBusy`.  I think the whole `try`/`catch` is redundant now.

File: x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java [hunk 2]
```diff
@@ -313,8 +310,22 @@ public void testHRDSplit() throws Exception {
...
+                assertThat(EntityUtils.toString(datafeedStatsResponse.getEntity()),
+                    containsString("\"state\":\"stopped\""));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }, 60, TimeUnit.SECONDS);
 
+        waitUntilJobIsClosed("hrd-split-job");
         try {
             assertBusy(() -> {
```
Comment (droberts195) [inline x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java:330]: Now that you're waiting for the job to be closed we shouldn't need to loop here.  By the time the job is closed it should have written all results it's ever going to write.

File: x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java [hunk 3]
```diff
@@ -313,8 +310,22 @@ public void testHRDSplit() throws Exception {
...
                 client().performRequest(new Request("POST", "/_refresh"));
```
Comment (droberts195) [inline x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java:331]: Now that our indices are hidden this won't be refreshing them.  The `assertBusy` will be masking this at the moment, but I think that should be removed and it's sometimes waiting a second unnecessarily.  We only need it to refresh `.ml-anomalies-*`, so might as well specify this in the refresh.

**Review by droberts195 — COMMENTED** [review]
If you replace line 337:

```
    if (responseBody.contains("\"count\":2")) {
```

with:

```
    assertThat(responseBody, containsString("\"count\":2"));
```

then the `else` clause on lines 359-362 can be removed.

File: x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java (hunk 1 above)
Comment (benwtrent) [reply x-pack/plugin/ml/qa/single-node-tests/src/test/java/org/elasticsearch/xpack/ml/transforms/PainlessDomainSplitIT.java:366]: For sure, I will refactor the test to make it cleaner

**Review by droberts195 — APPROVED** [review]
LGTM

**Comment by benwtrent** [comment]
@elasticmachine update branch

**Pull request merged**